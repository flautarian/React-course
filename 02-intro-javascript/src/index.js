console.log("Hello world!!!")

const dice = {
    faces: 20,
    getRandomValue: function(){
        // 👇️ get number between min (inclusive) and max (inclusive)
        return Math.floor(Math.random() * (this.faces+ 1));
    }
}

const diceResult =`${dice.getRandomValue()}`;
console.log(diceResult);