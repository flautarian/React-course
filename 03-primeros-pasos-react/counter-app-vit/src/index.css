html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    padding: 70px;
  }
  
  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
      monospace;
  }
  
button {
  padding: 15px;
  margin: 5px;
  font-size: 1.3em;
}