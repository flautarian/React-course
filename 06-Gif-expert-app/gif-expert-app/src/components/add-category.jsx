import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onSubmit = (event) => {
        event.preventDefault();
        if (!!inputValue) {
            onAddCategory(inputValue);
            setInputValue('');
        }
    }

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    return (
        <div className="center-content">
            <form onSubmit={ onSubmit } aria-label="form">
                <div>
                    <input
                        placeholder="get gifs like...."
                        value={inputValue}
                        onChange={ onInputChange }>
                    </input>
                </div>
                <button className="glow-on-hover" type="submit">
                    Add Gifs
                </button>
            </form>
        </div>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}

