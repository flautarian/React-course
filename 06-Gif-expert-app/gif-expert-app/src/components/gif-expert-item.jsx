import { Fragment } from "react";
import PropTypes from 'prop-types';
import { render, screen } from '@testing-library/react'

export const GifExpertItem = ({ title, url }) => {
    return (
        <Fragment>
            <div className="card">
                <img src={url} alt={title}></img>
                <p>{title}</p>
            </div>
        </Fragment>
    )
}

GifExpertItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
