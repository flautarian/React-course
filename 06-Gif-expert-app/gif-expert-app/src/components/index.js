export * from  './add-category'
export * from  './gif-expert-categories-grid'
export * from  './gif-expert-item'
