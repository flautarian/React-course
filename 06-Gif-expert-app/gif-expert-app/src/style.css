* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(211, 50, 239);
}

body {
    padding: 60px;
}

input {
    background-color: black;
    border-radius: 5px;
    border: 1px solid rgb(42, 0, 81);
    color: rgb(152, 152, 152);
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
}

input::placeholder {
    color: rgb(87, 87, 87);
}

.center-content {
    margin: 0;
    text-align: center;
}

.glow-on-hover {
    width: 120px;
    height: 25px;
    border: none;
    outline: none;
    color: #fff;
    background: #111;
    cursor: pointer;
    position: relative;
    z-index: 0;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.glow-on-hover:before {
    content: '';
    background: linear-gradient(45deg, #ff0000, #ff7300, #fffb00, #48ff00, #00ffd5, #002bff, #7a00ff, #ff00c8, #ff0000);
    position: absolute;
    top: -2px;
    left: -2px;
    background-size: 400%;
    z-index: -1;
    filter: blur(5px);
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    animation: glowing 20s linear infinite;
    opacity: 0;
    transition: opacity .3s ease-in-out;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.glow-on-hover:active {
    color: #000
}

.glow-on-hover:active:after {
    background: transparent;
}

.glow-on-hover:hover:before {
    opacity: 1;
}

.glow-on-hover:after {
    z-index: -1;
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background: #111;
    left: 0;
    top: 0;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

@keyframes glowing {
    0% {
        background-position: 0 0;
    }

    50% {
        background-position: 400% 0;
    }

    100% {
        background-position: 0 0;
    }
}

h1 {
    color: #cccccc;
}

h2 {
    font-size: 1.5rem;
    color: #cccccc;
}

h3 {
    font-size: 3rem;
    margin-bottom: 5px;
}

.card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
}

.card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
}

.card img {
    width: 100%;
}