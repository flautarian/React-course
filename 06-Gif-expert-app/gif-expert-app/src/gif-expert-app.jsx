import { useState, Fragment } from "react";
import { AddCategory, GifExpertCategoriesGrid } from "./components/index";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'One piece']);

    const onAddCategory = (value) => {
        console.log(value);
        if (!!value && !categories.includes(value) )
            setCategories([value, ...categories]);
    }

    return (
        <Fragment>
            {/* Title */}
            <h1>Gif Expert App</h1>
            {/* Input */}
            <AddCategory onAddCategory={onAddCategory} />
            {/* Results */}
            {categories.map(category => {
                return (
                    <GifExpertCategoriesGrid key={category} category={ category }/>
                );
            })}
            
            
        </Fragment>
    )
}