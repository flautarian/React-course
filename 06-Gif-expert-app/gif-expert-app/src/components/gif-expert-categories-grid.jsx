import { Fragment } from "react";
import { GifExpertItem } from "./gif-expert-item";
import { UseFetchGifs } from "../hooks/UseFetchGifs";
import PropTypes from 'prop-types';

export const GifExpertCategoriesGrid = ({ category }) => {
    
    const { images, isLoading } = UseFetchGifs( category );
    
    return (
        <Fragment>
            <h3>{category}</h3>
            {
                 isLoading && ( <h2>Loading...</h2> )
            }
            <div className="card-grid">    
                {images.map(image => {
                    return (
                        <GifExpertItem key={image.id} {...image} />
                    );
                })}
            </div>
        </Fragment>
    )
}

GifExpertCategoriesGrid.propTypes = {
    category: PropTypes.string.isRequired
}