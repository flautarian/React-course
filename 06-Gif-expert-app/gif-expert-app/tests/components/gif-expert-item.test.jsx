import { render, screen } from '@testing-library/react'
import { GifExpertItem } from '../../src/components';

describe('gif-expert-item-test', () => { 

    const testImage = {
        id: 1,
        url: "test-url",
        title: "title-test"
    }
    
    test('should match snapshot', () => {        
        const { container } = render( <GifExpertItem key={testImage.id} {...testImage}/> );
        expect( container ).toMatchSnapshot();
    });

    test('should render url image', () => {
        render( <GifExpertItem key={testImage.id} {...testImage}/> );
        expect( screen.getByRole( 'img' ).src).toContain(testImage.url);
        expect( screen.getByRole( 'img' ).alt).toBe(testImage.title);
    });
    
    test('should render title', () => { 
        render( <GifExpertItem key={testImage.id} {...testImage}/> );
        expect( screen.getByText( testImage.title )).toBeTruthy();
     })

})