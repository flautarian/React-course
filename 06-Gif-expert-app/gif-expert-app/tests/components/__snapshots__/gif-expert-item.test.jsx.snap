// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gif-expert-item-test should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title-test"
      src="test-url"
    />
    <p>
      title-test
    </p>
  </div>
</div>
`;
